src/com/sun/jna/Pointer.java

```diff
@@ -1271,22 +1273,27 @@ public void setString(long offset, String value, String encoding) {
     
     /** Dump memory for debugging purposes. */
     public String dump(long offset, int size) {
-        String LS = System.getProperty("line.separator");
-        String contents = "memory dump" + LS;
         final int BYTES_PER_ROW = 4;
-        byte[] buf = getByteArray(offset, size);
-        for (int i=0;i < buf.length;i++) {
-            if ((i % BYTES_PER_ROW) == 0) contents += "[";
-            if (buf[i] >=0 && buf[i] < 16)
-                contents += "0";
-            contents += Integer.toHexString(buf[i] & 0xFF);
-            if ((i % BYTES_PER_ROW) == BYTES_PER_ROW-1 && i < buf.length-1)
-                contents += "]" + LS;
-        }
-        if (!contents.endsWith("]" + LS)) {
-            contents += "]" + LS;
-        }
-        return contents;
+        final String TITLE = "memory dump";
+        // estimate initial size assuming a 2 char line separator
+        StringWriter sw = new StringWriter(TITLE.length() + 2 + size * 2 + (size / BYTES_PER_ROW * 4));
+        PrintWriter out = new PrintWriter(sw);
+        out.println(TITLE);
+//        byte[] buf = getByteArray(offset, size);
```

Comment:
That goes back to the question in my original comment of this pull request, wether it's better to copy the buffer into a byte[] or to directly access the buffer. I tend to think that accessing the buffer is better, so based on that the commented lines could be removed.